pytorch_lightning/trainer/connectors/optimizer_connector.py

```diff
@@ -36,6 +36,9 @@ def update_learning_rates(self, interval: str, monitor_metrics=None):
             return
 
         for scheduler_idx, lr_scheduler in enumerate(self.trainer.lr_schedulers):
+            if opt_indices and lr_scheduler['opt_idx'] and lr_scheduler['opt_idx'] not in opt_indices:
```

Comment:
```suggestion
            if lr_scheduler['opt_idx'] not in opt_indices:
```
I think if opt_indices is an empty list then we can just return.... a few lines above.

Reply:
I'm explicitly checking for the conditions together in order to maintain backwards compatibility. Any other call to `update_learning_rates` (in previous versions) without setting `opt_indices` will prevent the LR schedulers from being updated if we return early.

Reply:
not sure what you meant here. in previous version there won't be any opt_indices but there won't be any of these checks either. you added this to fix this bug. I think with this fix `opt_indices` can never be an empty list. Updated the suggestion above.

Reply:
I mean if there were any calls to `update_learning_rates` in a custom codebase separately and if their PL version was to be updated, returning early due to an empty `opt_indices` would break things. If you think that's unlikely, I can make an empty list as the default.

Reply:
yeah... I'd suggest an empty list for now